examples/channelHelper.js

```diff
@@ -0,0 +1,218 @@
+class ChannelHelper {
+    static IngestionMode = {
+        // P2P (no ingestion)
+        OFF: 0,
+
+        // Do not call DescribeMediaStorageConfiguration API, assume channel
+        // is already configured for ingestion.
+        ON: 1,
+
+        // Call DescribeMediaStorageConfiguration API to dynamically determine
+        // operating mode.
+        DETERMINE_THROUGH_DESCRIBE: 2,
+    };
+
+    constructor(channelName, clientArgs, endpoint, role, ingestionMode, loggingPrefix) {
+        this._channelName = channelName;
+        this._clientArgs = clientArgs;
+        this._role = role;
+        this._endpoint = endpoint;
+        this._ingestionMode = ingestionMode;
+        this._loggingPrefix = loggingPrefix;
+    }
+
+    // Must be called first
+    // Creates all the clients used to interface with Kinesis Video Streams Signaling
+    async init() {
+        await this._initializeClients();
+    }
+
+    // Returns the Signaling Channel ARN
+    // Only available after init()
+    getChannelArn = () => {
+        return this._channelArn;
+    };
+
+    // Returns the KinesisVideo client
+    // Used to invoke APIs under https://docs.aws.amazon.com/kinesisvideostreams/latest/dg/API_Operations_Amazon_Kinesis_Video_Streams.html
+    // Only available after init()
+    getKinesisVideoClient = () => {
+        return this._kinesisVideoClient;
+    };
+
+    // Returns the Signaling WebSocket client
+    // Used to invoke APIs under https://docs.aws.amazon.com/kinesisvideostreams-webrtc-dg/latest/devguide/kvswebrtc-websocket-apis.html
+    // Only available after init()
+    getSignalingClient = () => {
+        return this._signalingClient;
+    };
+
+    // Returns true if this is in ingestion mode
+    // If this was initialized with DETERMINE_THROUGH_DESCRIBE, after init() is
+    // called, this will be updated accordingly
+    isIngestionEnabled = () => {
+        return this._ingestionMode === ChannelHelper.IngestionMode.ON;
+    };
+
+    // Returns the Kinesis Video WebRTC Storage Client
+    // Used to invoke APIs under https://docs.aws.amazon.com/kinesisvideostreams/latest/dg/API_Operations_Amazon_Kinesis_Video_WebRTC_Storage.html
+    // Only available after init()
+    getWebRTCStorageClient = () => {
+        return this._webrtcStorageClient;
+    };
+
+    // Returns the Kinesis Video Stream ARN that this channel's media is ingested to
+    // Only available after init(), and if this was initialized with DETERMINE_THROUGH_DESCRIBE
+    // and the DescribeMediaStorageConfiguration API returned that this channel is configured for ingestion
+    getStreamArn = () => {
+        return this._streamArn;
+    };
+
+    // Fetch and return TURN servers
+    // Only available after init()
+    fetchTurnServers = async () => {
```

Comment:
Unrelated to the PR: But these APIs would help the viewer too right?

Reply:
Yep - I will integrate that at a later point. Taking small steps here